src/UI/Internal/Utils/Element.elm

```diff
@@ -56,29 +58,31 @@ id value =
         |> Element.htmlAttribute
 
 
-overlayBackground : msg -> Element msg
-overlayBackground onClickMsg =
+clickElsewhereToLeave : msg -> List (Attribute msg) -> Element msg -> Element msg
+clickElsewhereToLeave onClickMsg backgroundStyleAttributes foregroundContent =
+    let
+        backgroundAttributes =
+            positionFixed
+                -- Needs for starting at the top-left corner
+                :: zIndex 8
```

Comment:
Will this zindex work for any page? Maybe it should be passed in and stored as a constant like I suggested previously. To be more specific, giving it a name like mySpecificElementsNameZIndex, not zIndex8